cli/azd/internal/repository/app_init.go

```diff
@@ -560,3 +504,77 @@ func (i *Initializer) prjConfigFromDetect(
 
 	return config, nil
 }
+
+// ServiceFromDetect creates a ServiceConfig from an appdetect project.
+func ServiceFromDetect(
+	root string,
+	svcName string,
+	prj appdetect.Project) (project.ServiceConfig, error) {
+	svc := project.ServiceConfig{
+		Name: svcName,
+	}
+	rel, err := filepath.Rel(root, prj.Path)
+	if err != nil {
+		return svc, err
+	}
+
+	if svc.Name == "" {
+		dirName := filepath.Base(rel)
+		if dirName == "." {
+			dirName = filepath.Base(root)
+		}
+
+		svc.Name = names.LabelName(dirName)
+	}
+
+	svc.Host = project.ContainerAppTarget
+	svc.RelativePath = rel
+
+	language, supported := LanguageMap[prj.Language]
+	if !supported {
+		return svc, fmt.Errorf("unsupported language: %s", prj.Language)
+	}
+
+	svc.Language = language
+
+	if prj.Docker != nil {
+		relDocker, err := filepath.Rel(prj.Path, prj.Docker.Path)
+		if err != nil {
+			return svc, err
+		}
+
+		svc.Docker = project.DockerProjectOptions{
+			Path: relDocker,
+		}
+	}
+
+	if prj.HasWebUIFramework() {
+		// By default, use 'dist'. This is common for frameworks such as:
+		// - TypeScript
+		// - Vite
+		svc.OutputPath = "dist"
+
+	loop:
```

Comment:
NIT: I think that with the refactoring to extract this in to it's own function, you can now remove the label and just call `break`?